src/App.tsx

```diff
@@ -26,70 +27,125 @@ type ReorderOptions = {
+        return firstGood.length - secondGood.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+export class App extends React.Component<{}, ReorderOptions> {
+  state: ReorderOptions = {
+    sortType: SortType.NONE,
+    isReversed: false,
+  };
+
+  getCorrectMethod(name: string) {
+    switch (name) {
+      case 'reverse':
+        this.setState(state => ({
+          isReversed: !state.isReversed,
+        }));
+        break;
+
+      case 'sortByLength':
+        this.setState({ sortType: SortType.LENGTH });
+        break;
+
+      case 'sortByAlp':
+        this.setState({ sortType: SortType.ALPHABET });
+        break;
+
+      case 'reset':
+        this.setState({
+          sortType: SortType.NONE,
+          isReversed: false,
+        });
+        break;
+
+      default:
+        throw new Error('Invalid action name');
+    }
+  }
+
+  render() {
+    const { sortType, isReversed } = this.state;
+    const visibleGoods = getReorderedGoods(goodsFromServer, this.state);
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={sortType === SortType.ALPHABET
+              ? 'button is-info'
+              : 'button is-info is-light'}
+            onClick={() => this.getCorrectMethod('sortByAlp')}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={sortType === SortType.LENGTH
+              ? 'button is-success'
+              : 'button is-success is-light'}
+            onClick={() => this.getCorrectMethod('sortByLength')}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={cn('button', 'is-warning', {
+              'is-light': !isReversed,
+            })}
+            onClick={() => this.getCorrectMethod('reverse')}
+          >
+            Reverse
+          </button>
+
+          {isReversed || sortType !== SortType.NONE
```

Comment:
move to the variable
```suggestion
          {...
```